.github/workflows/static-checks.yml

```diff
@@ -56,6 +56,12 @@ jobs:
         run: |
           echo "JAVA_HOME=$JAVA_HOME_${{ env.java_version }}_X64" >> $GITHUB_ENV
 
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: maven-static-checks-${{ matrix.java }}-${{ hashFiles('**/pom.xml') }}
```

Comment:
Is there a reason to have these jobs all use different cache keys -- could/should they share a cache?

Reply:
I added the job name because I didn't wanted to risk in case that cause any problems; right now for every PR every artifact is downloaded  >7 times or something...reducing that removing all the static-checks jobs removes around 4 - because it will be cached based on the pom.xml hashes

we could improve on this - but the reusable test builder jobs will still download everything from 0; and since caching there is used to distribute artifacts - the situation is a bit more complicated :D